Arrangement_on_surface_2/include/CGAL/Aos_observer.h

```diff
@@ -0,0 +1,573 @@
+
+  /*! Notification before the arrangement is cleared. */
+  virtual void before_clear() {}
+
+  /*! Notification after the arrangement is cleared. */
+  virtual void after_clear() {}
+
+  /*! Notification before a global operation modifies the arrangement. */
+  virtual void before_global_change() {}
+
+  /*! Notification after a global operation is completed. */
+  virtual void after_global_change() {}
+  //@}
+
+  /// \name Notification functions on observer attachment or detachment.
+  //@{
+
+  /*! Notification before the observer is attached to an arrangement.
+   * \param arr The arrangement we are about to attach the observer to.
+   */
+  virtual void before_attach(const Arrangement_2& /* arr */) {}
+
+  /*! Notification after the observer has been attached to an arrangement. */
+  virtual void after_attach() {}
+
+  /*! Notification before the observer is detached from the arrangement. */
+  virtual void before_detach() {}
+
+  /*! Notification after the observer has been detached to the arrangement. */
+  virtual void after_detach() {}
+  //@}
+
+  /// \name Notification functions on local changes in the arrangement.
+  //@{
+
+  /*!
+   * Notification before the creation of a new vertex.
+   * \param p The point to be associated with the vertex.
+   *          This point cannot lies on the surface boundaries.
+   */
+  virtual void before_create_vertex(const Point_2& /* p */) {}
+
+  /*! Notification after the creation of a new vertex.
+   * \param v A handle to the created vertex.
+   */
+  virtual void after_create_vertex(Vertex_handle /* v */)
+  {}
+
+  /*! Notification before the creation of a new boundary vertex.
+   * \param p The on the surface boundary.
+   * \param ps_x The boundary condition of the vertex in x.
+   * \param ps_y The boundary condition of the vertex in y.
+   */
+  virtual void before_create_boundary_vertex(const Point_2& /* p */,
+                                             Arr_parameter_space /* ps_x */,
+                                             Arr_parameter_space /* ps_y */)
+  {}
+
+  /*! Notification before the creation of a new boundary vertex.
+   * \param cv The curve incident to the surface boundary.
+   * \param ind The relevant curve-end.
+   * \param ps_x The boundary condition of the vertex in x.
+   * \param ps_y The boundary condition of the vertex in y.
+   */
+  virtual void before_create_boundary_vertex(const X_monotone_curve_2& /* cv */,
+                                             Arr_curve_end /* ind */,
+                                             Arr_parameter_space /* ps_x */,
+                                             Arr_parameter_space /* ps_y */)
+  {}
+
+  /*! Notification after the creation of a new vertex at infinity.
+   * \param v A handle to the created vertex.
+   */
+  virtual void after_create_boundary_vertex(Vertex_handle /* v */) {}
+
+  /*! Notification before the creation of a new edge.
+   * \param c The x-monotone curve to be associated with the edge.
```

Comment:
`x-monotone` elsewhere it is like a formula `\f$ x\f$-monotone` (also at other places)